File: core/src/com/symbol/game/ecs/component/TextureComponent.java
package com.symbol.game.ecs.component;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.utils.Pool;

public class TextureComponent implements Component, Pool.Poolable {

    public TextureRegion texture = null;
    public String textureStr = null;

    @Override
    public void reset() {
        texture = null;
        textureStr = null;
    }
}

File: core/src/com/symbol/ecs/component/TextureComponent.kt
package com.symbol.ecs.component

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.utils.Pool

class TextureComponent : Component, Pool.Poolable {

    var texture: TextureRegion? = null
    var textureStr: String? = null

    override fun reset() {
        texture = null
        textureStr = null
    }
}